openhands/integrations/provider.py

```diff
@@ -325,20 +325,26 @@ def get_provider_env_key(cls, provider: ProviderType) -> str:
     async def verify_repo_provider(
         self, repository: str, specified_provider: ProviderType | None = None
     ) -> Repository:
+        errors = []
+
         if specified_provider:
             try:
                 service = self._get_service(specified_provider)
                 return await service.get_repository_details_from_repo_name(repository)
-            except Exception:
-                pass
+            except Exception as e:
+                errors.append(f'{specified_provider.value}: {str(e)}')
 
         for provider in self.provider_tokens:
             try:
                 service = self._get_service(provider)
                 return await service.get_repository_details_from_repo_name(repository)
-            except Exception:
-                pass
+            except Exception as e:
+                errors.append(f'{provider.value}: {str(e)}')
 
+        # Log all accumulated errors before raising AuthenticationError
+        logger.error(
+            f'Failed to access repository {repository} with all available providers. Errors: {"; ".join(errors)}'
```

Comment:
This is probably not going to show stacktrace, but is certainly better than the missing logs we had before.

Reply:
Yep just want the error code to surface here, not concerned about the stack trace for this particular failure point